rpcs3/main.cpp

```diff
@@ -130,8 +130,9 @@ LOG_CHANNEL(q_debug, "QDEBUG");
 		dlg.exec();
 	};
 
-#ifdef __APPLE__
+#if defined(__APPLE__) && defined(BLOCKS) // BLOCKS is required for dispatch_sync, but GCC-11 does not support it
 	// Cocoa access is not allowed outside of the main thread
```

Comment:
Can you pass `[]{ show_report(text); }` lambda instead?

Reply:
He can't compile it, that's why he added the ifdef for BLOCK.
Or can you confirm it compiles?

Reply:
I'm just wondering if lambdas can be used instead of alien ^{} syntax, I think I saw some examples in wild.

Reply:
We won't know in this PR, as the author uses gcc11